### ISSUE TITLE
Documentation of `Controller::distToSegmentSquared()` incorrect

### ISSUE THREAD

**Issue by lewie-donckers** [nobleo/path_tracking_pid#128]
While working on unit tests for `Controller::distToSegmentSquared()` I noticed the following.

The documentation suggests that the pose it returns is the point on the segment PV (start, end) that is closest to W (point). It does not do this.

I had the following test case:
```cpp
TEST(PathTrackingPidCalculations, ClosestPointOnSegment_CloseToStart)
{
  const auto start = create_transform(2, 2, 0);
  const auto end = create_transform(4, 4, 0);
  const auto point = create_transform(0, 1, 0);
  const auto ref = start;

  EXPECT_EQ(ref, closestPointOnSegment(start, end, point, false));
}
```

`point` is closest to `start` of any point on `start`-`end` but the following is returned: `[1.759, 2.319, 0]`.

**Comment by rokusottervanger** [comment]
Looks like a nice catch to me.

@cesar-lopez-mar is this expected behavior?

**Comment by lewie-donckers** [comment]
Oops. Might be false alarm. We messed up the documentation and because of that I messed up the order of the arguments in my unit tests. Will fix the documentation and tests and see what happens tomorrow.

**Comment by rokusottervanger** [comment]
Ah, I did the math by hand for this test case. It should be exactly `[1.760, 2.320, 0]`, so that checks out.

**Comment by lewie-donckers** [comment]
It's just a documentation "bug". I fixed it in PR #133 .
